Tests/Integration/LnsProtocolMessageProcessorConcentratorDeduplicationIntegrationTests.cs

```diff
@@ -22,11 +22,13 @@ namespace LoRaWan.Tests.Integration
     /// <summary>
     /// These tests test the integration between <see cref="LnsProtocolMessageProcessor"/> and <see cref="ConcentratorDeduplication"/>.
     /// </summary>
-    public sealed class LnsProtocolMessageProcessorConcentratorDeduplicationIntegrationTests
+    public sealed class LnsProtocolMessageProcessorConcentratorDeduplicationIntegrationTests : IDisposable
     {
-        private static async Task<TestServer> TestServerAsyncFactory(Mock<IMessageDispatcher> messageDispatcherMock)
+        private IHost testHost;
+
+        private async Task CreateTestHostAsync(Mock<IMessageDispatcher> messageDispatcherMock)
```

Comment:
Wonder why you changed the return type here? I think it's unrelated and would even give fewer changes in the PR.

Reply:
I think it's a bit clearer this way because what gets created from the method is also stored in the state to get stopped/disposed later on. I consider this a nit and will resolve - let me know if you feel otherwise.